website/content/en/docs/best-practices/multi-tenancy.md

```diff
@@ -0,0 +1,162 @@
+For this reason, if you are integrating with [OLM][olm], you should, **always ship your operator `Deployment` with resource requests amounts for CPU and memory**.
+
+The [ResourceQuota][k8s-resources-quotas] configuration also is applied to tenant workloads that are managed by your operator. Cluster administrators will typically set
+a [ResourceQuota][k8s-resources-quotas] for each tenant's namespace/project as part of the onBoarding. If your operator creates `Pods/Deployments` inside
+the tenant namespace which do not specify at least resource requests for `CPU` and `Memory`, those `Pods/Deployments` will fail to schedule and tenants
+will not be able to use your operator. Therefore, your operator should **always apply at least resource requests for CPU and
+memory** for `Pods/Deployments` that it creates as part of reconciliation. Ideally your operator also applies resource 
+limits to those `Pods/Deployments`. 
+
+#### General guidelines
+
+**Note that pretty much all controllers consume:**
+- `CPU`: largely based on the number of reconciliations they perform, which are generally related to event activity for resources they're watching.
+- `Memory`: largely based on the number of primary resources that exist (multiplied by some factor based on the number of operand resources they need to watch as a result) via informer caches.
+
+Operator authors cannot anticipate how they scale in usage in all customer environments, so setting one-size-fits-all 
+that fits well for all scenarios is not possible then, clusters administrators must be allowed to change the values 
+in order to respond to changes in their clusters’ needs, such as by resizing control plane nodes to provide more resources.
+
+**Following are some general recommendations to manage the resources:**
+
+- **MUST** declare resource requests for both CPU and memory.
+- **OPTIONALLY** setting the resources limit for Memory and CPU (Note that if your project is integrated with OLM then, admins would be able to customize it via the [subscription][olm-subscriptions]). Also, be aware that our project was built with Kubebuilder or SDK scaffold then for the `Operator/manager`(See `config/manager/manager.yaml`), then some values which allow you to get started are scaffolded by default, however, you ought to optimize based on the needs.
+- **Should** provide the mechanisms for [Monitoring compute & memory resource usage][k8s-metrics]. Kubebuilder and SDK provide some [metrics][kb-metrics] which can help you with. Note that cluster admins can use it to monitor and resize the Operator and its Operands.
+- **OPTIONALLY** ensuring that the Operators uses [liveness health check][k8s-probes] which can be used to check the `CPU` performance. If your project was built with Kubebuilder or SDK scaffold then, it is scaffolded by default for the `Operator/manager`. (See `config/manager/manager.yaml`).
```

Comment:
"CPU performance" is probably not the right wording. Assuming "CPU usage" I don't think that a liveness probe is a good mechanism for that. The primary use case of a liveness probe is to catch a deadlock: an unrecoverable state. It then restarts the pod with a "virgin" state, which may allow to "work around" a bug.
Two things:
- If the liveness probe is too aggressive it may cause the pod to get restarted when it is just slow to respond due  to its load. This is counterproductive and at the opposite effect of what is desirable
- With controllers most the state should be stored in custom resources, which means that in some cases (possibly not all) restarting the pod won't help with the deadlock
- 
Long story short I would recommend "considering" rather than "ensuring" the use of health checks

Reply:
Reworded to `which can be used to check the performance to know more about the CPU usage. ` and used the considering

Reply:
I think we can remove the probe from here.